mobile/src/containers/Settings.jsx

```diff
@@ -31,8 +31,17 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
     dispatch(unlink(client))
     ownProps.router.replace('/onboarding')
   },
-  setBackupImages: (e) => dispatch(setBackupImages(e.target.checked)),
-  setWifiOnly: (e) => dispatch(setWifiOnly(e.target.checked)),
+  setBackupImages: async (e) => {
+    const value = e.target.checked
+    const action = await dispatch(backupImages(value))
+    if (value === true && value !== action) {
+      e.target.checked = false
```

Comment:
when we call `backupImages` may be user don't authorized files access. So we ask him. If he refuses, the state don't change. Checkbox and state are different so we uncheck it ;)